packages/core-utils/src/logic-utils.ts

```diff
@@ -0,0 +1,58 @@
+export const DEFAULT = Symbol('Default switch case');
+
+type CondCase<T> = [boolean, (() => T) | T];
+type CondDefault<T> = [typeof DEFAULT, (() => T) | T] | (() => T) | T;
+
+type Cond = {
+  <T>(...args: [...CondCase<T>[], CondDefault<T>]): T;
+  <T>(...args: CondCase<T>[]): T | undefined;
+};
+
+const maybeCall = <T>(arg: (() => T) | T): T => {
+  return arg instanceof Function ? arg() : arg;
+};
+
+/**
+ * Takes any number of [predicate, factory] cases, followed by an optional default factory.
+ * Returns factory() for the first truthy predicate, otherwise returns the default factory().
+ * Returns undefined if no predicate matches and there is no default factory.
+ */
+export const cond: Cond = (...args) => {
+  for (const arg of args) {
+    if (Array.isArray(arg)) {
```

Comment:
What, the entries don't have to be of the form [predicate, factory] but instead can just be "factory"? I don't see any test cases that cover this usage.

Reply:
The default case is just "factory". The type declaration requires that a default case like that can only be given as the last argument. This test case covers it:
```ts
it("returns result of default value factory if one is provided and no case's condition is true", () => {
    // Act
    const result = cond([false, () => 'a'], [false, () => 'b'], [false, () => 'c'], () => 'z');

    // Assert
    expect(result).toBe('z');
  });
```